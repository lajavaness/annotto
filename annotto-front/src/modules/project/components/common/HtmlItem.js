import PropTypes from 'prop-types'
import React from 'react'

import { sanitizer } from 'shared/utils/htmlUtils'

import * as Styled from './__styles__/HtmlItem.styles'

const HtmlItem = ({ content }) => {
  return (
    <Styled.Root data-testid={'__html-item__'}>
      <Styled.Content dangerouslySetInnerHTML={{ __html: sanitizer(content) }} />
    </Styled.Root>
  )
}

export default HtmlItem

HtmlItem.propTypes = {
  /** Defines the content of the item. */
  content: PropTypes.string,
}

HtmlItem.defaultProps = {
  content: null,
}
