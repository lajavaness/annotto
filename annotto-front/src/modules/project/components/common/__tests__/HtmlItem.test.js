import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import HtmlItem from 'modules/project/components/common/HtmlItem'

const defaultProps = {}
const getInstance = (props = {}) => (
  <ThemeProvider theme={theme}>
    <HtmlItem {...defaultProps} {...props} />
  </ThemeProvider>
)

describe('HtmlItem', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(getInstance())
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders content', () => {
    const content = '<div data-testid="__foo__">Foo</div>'
    const { getByTestId } = render(getInstance({ content: `${content}` }))
    expect(getByTestId('__foo__')).toBeVisible()
  })
})
