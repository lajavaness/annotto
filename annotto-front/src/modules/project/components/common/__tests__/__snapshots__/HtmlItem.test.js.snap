// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-grxQYx iTBcUt"
    data-testid="__html-item__"
  >
    <div
      class="sc-ZqGJI"
    />
  </div>
</DocumentFragment>
`;
